homeassistant/components/modbus/validators.py

```diff
@@ -280,6 +299,13 @@ def check_hvac_target_temp_registers(config: dict) -> dict:
         wrn = f"{CONF_FAN_MODE_REGISTER} overlaps CONF_TARGET_TEMP register(s). {CONF_FAN_MODE_REGISTER} is not loaded!"
         _LOGGER.warning(wrn)
         del config[CONF_FAN_MODE_REGISTER]
+    if (
+        CONF_SWING_MODE_REGISTER in config
+        and config[CONF_SWING_MODE_REGISTER][CONF_ADDRESS] in config[CONF_TARGET_TEMP]
+    ):
+        wrn = f"{CONF_SWING_MODE_REGISTER} overlaps CONF_TARGET_TEMP register(s). {CONF_SWING_MODE_REGISTER} is not loaded!"
```

Comment:
You need to test with all addresses in the configuration (check_config).

What if one is a list and the other an int.

Reply:
@janiversen I was managing this request related to the function `def check_hvac_target_temp_registers`, function written at the time when no central validation of the modbus entities existed. I wrote that function to check if the new target temperature registers overlap with some.. 
Now, instead to fix that, I was checking how your function `def validate_entity` works, because **in my opinion** it could be the best way to manage also the conflicts inside the same entity. 
I found 2 main problems: 
1) that function doesn't consider the possibility to have `[CONF_FAN_MODE_REGISTER][CONF_ADDRESS]` as list and `[CONF_TARGET_TEMP]` as list. 
2) it doesn't recognize conflicting registers inside the same entity. I tested this also in my production server, updated at the last core. 

I may suggest to work only with  `def validate_entity` fixing the bugs and removing my previous function `def check_hvac_target_temp_registers` in order to simplify the code. But as maintainer, I kindly ask to you how you want to proceed. Thanks. 
Let me know.

Reply:
Check_config existed when you made your first modbus contribution, but you might have overseen it. the validate functions equally existed (but are now no longer inline functions).

This PR is about swing mode, so please do not mix a lot of other changes in that, make a new PR instead.

Mid term I am removing all voluptuous validators and add them to check_config, since this allows to create an issue and thus be a lot more user friendly.

Reply:
Ok. So, at this time I'll improve `def check_hvac_target_temp_registers` only.. Then, we'll see.

Reply:
Fixed the case where swing_mode register is a list. (https://github.com/home-assistant/core/pull/113710/commits/aca776aced268ade5f74a2671f6f3c976d50145f). `CONF_TARGET_TEMP` is always a list. 
The goal of the function was to test only if `CONF_TARGET_TEMP` is overlapping with some other registers.. 
This function and `validate_entity` will be fixed in a another PR to check registers conflicting into the whole entity.